pkg/porter/credentials.go

```diff
@@ -35,48 +35,51 @@ func (l CredentialsFileList) Less(i, j int) bool {
 	return l[i].Modified.Before(l[j].Modified)
 }
 
-// ListCredentials lists credentials using the provided printer.PrintOptions
-func (p *Porter) ListCredentials(opts printer.PrintOptions) error {
-	dir, err := p.Config.GetCredentialsDir()
+// fetchCredentials fetches all credentials from the designated credentials dir
+func (p *Porter) fetchCredentials() (*CredentialsFileList, error) {
+	credsDir, err := p.Config.GetCredentialsDir()
 	if err != nil {
-		return errors.Wrap(err, "unable to determine credentials directory")
-	}
-
-	// TODO: should this go somewhere else?
-	// i.e., where should logic for init-ing dirs needed by porter go?
-	// e.g., on 'porter creds list' with a fresh porter install/home
-	if ok, _ := p.Context.FileSystem.DirExists(dir); !ok {
-		p.Context.FileSystem.Mkdir(dir, os.ModePerm)
+		return &CredentialsFileList{}, errors.Wrap(err, "unable to determine credentials directory")
 	}
 
 	credentialsFiles := CredentialsFileList{}
-	err = p.Context.FileSystem.Walk(dir, func(path string, info os.FileInfo, err error) error {
-		if !info.IsDir() {
-			credSet := &credentials.CredentialSet{}
-			data, err := p.Context.FileSystem.ReadFile(path)
-			if err != nil {
-				return errors.Wrap(err, fmt.Sprintf("unable to load credential set from %s:\n%s", path, err))
-			}
-			if err = yaml.Unmarshal(data, credSet); err != nil {
-				return errors.Wrap(err, "unable to unmarshal credential set")
+	if ok, _ := p.Context.FileSystem.DirExists(credsDir); ok {
+		err = p.Context.FileSystem.Walk(credsDir, func(path string, info os.FileInfo, err error) error {
+			if !info.IsDir() {
+				credSet := &credentials.CredentialSet{}
+				data, err := p.Context.FileSystem.ReadFile(path)
+				if err != nil {
+					return errors.Wrap(err, fmt.Sprintf("unable to load credential set from %s:\n%s", path, err))
+				}
+				if err = yaml.Unmarshal(data, credSet); err != nil {
+					return errors.Wrap(err, "unable to unmarshal credential set")
+				}
+				credentialsFiles = append(credentialsFiles,
+					CredentialsFile{Name: credSet.Name, Modified: info.ModTime()})
 			}
-			credentialsFiles = append(credentialsFiles,
-				CredentialsFile{Name: credSet.Name, Modified: info.ModTime()})
+			return nil
+		})
+		if err != nil {
+			return &CredentialsFileList{}, errors.Wrap(err, "encountered error while listing credentials")
```

Comment:
It would be really frustrating if there was a problem with a single credential, and that preventing anything from being printed out.

I would lean towards (maybe in a follow up PR), is when it's an error reading or unmarshaling a credential, print the error to `p.Err`, potentially wrapped behind a p.Debug check. That way they could see the see the problem, and filtering it using 2>, and not have that prevent them from seeing their other credentials.

Reply:
Good point.  I've updated the logic to log to `p.Err` if debug is set for both unmarshal and `ReadFile` failures and return `nil` in both cases, such that listing may proceed.  Added a test as well.  How does this look?